### PR TITLE
r: build framework based install again

### PR DESCRIPTION
**Pull request by JCount**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
build framework based install again

### CODE REVIEW

File: Formula/r.rb [hunk 1]
```diff
@@ -11,6 +12,7 @@ class R < Formula
   end
 
   depends_on "pkg-config" => :build
+  depends_on "texinfo" => :build
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: What is this needed for?

File: Formula/r.rb [hunk 2]
```diff
@@ -30,6 +32,8 @@ def install
     args = [
       "--prefix=#{prefix}",
       "--enable-memory-profiling",
+      "--enable-R-framework=#{frameworks}",
+      "--disable-java",
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: What is this needed for?

File: Formula/r.rb [hunk 3]
```diff
@@ -64,20 +68,37 @@ def install
...
+
+    bin.install_symlink Dir[r_home/"bin/{R,Rscript}"]
+    include.install_symlink Dir[r_home/"include/*"]
+    lib.install_symlink Dir[r_home/"lib/*"]
+    (lib/"pkgconfig").install_symlink frameworks/"lib/pkgconfig/libR.pc"
+    man1.install_symlink Dir[r_home/"man1/*"]
   end
 
-  def post_install
+  def site_library
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: Make this a variable instead of a method.

File: Formula/r.rb [hunk 4]
```diff
@@ -64,20 +68,37 @@ def install
...
+    ENV.delete "R_HOME"
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: Did you find this to be needed for something?

File: Formula/r.rb [hunk 5]
```diff
@@ -64,20 +68,37 @@ def install
...
     short_version =
       `#{bin}/Rscript -e 'cat(as.character(getRversion()[1,1:2]))'`.strip
-    site_library = HOMEBREW_PREFIX/"lib/R/#{short_version}/site-library"
+    HOMEBREW_PREFIX/"lib/R/#{short_version}/site-library"
+  end
+
+  def site_library_cellar
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: Make this a variable instead of a method.

File: Formula/r.rb [hunk 6]
```diff
@@ -64,20 +68,37 @@ def install
...
+    frameworks/"R.framework/Resources/site-library"
+  end
+
+  def post_install
     site_library.mkpath
-    ln_s site_library, lib/"R/site-library"
+    site_library_cellar.unlink if site_library_cellar.exist?
```
Comment (MikeMcQuaid) [inline Formula/r.rb]: When would this exist?

**Review by MikeMcQuaid — CHANGES_REQUESTED** [review]

**Comment by MikeMcQuaid** [comment]
Can we add a reference in the commit message to the relevant issues/PRs for this? Thanks!

**Review by jschueller — CHANGES_REQUESTED** [review]
the test section is a bit light to prevent regressions like #15717

At least the R CMD should be tested, like
R CMD BATCH -h

else you can see
https://github.com/conda-forge/r-base-feedstock/blob/master/recipe/meta.yaml

**Comment by MikeMcQuaid** [comment]
@jschueller `-h` doesn't seem like a useful test; can you suggest an invocation that will do something? Thanks!

**Comment by jschueller** [comment]
At least it will test that it's present, which is not currently the case. 
maybe "R CMD config DYLIB_EXT" ?
https://cran.r-project.org/doc/manuals/r-release/R-intro.html

**Pull request closed without merging**